rustls/src/client/tls12.rs

```diff
@@ -174,6 +174,7 @@ mod server_hello {
                         Ok(Box::new(ExpectNewTicket {
                             cert_verified,
                             config,
+                            peer_identity: resuming.peer_identity().clone(),
```

Comment:
There seems to be quite a bit of extra cloning in this commit, which feels ugly (even though it does not seem to have an effect on performance). Where does that come from? It doesn't seem obvious to me.